NetSSL_OpenSSL/src/SecureSocketImpl.cpp

```diff
@@ -300,6 +306,9 @@ void SecureSocketImpl::shutdown()
 			int rc = ::SSL_shutdown(_pSSL);
 #endif
 			if (rc < 0) handleError(rc);
+
+			l.unlock();
```

Comment:
Why is mutex unlocked explicitly before end of scope only in this function and not also others?